tools/resource-gen/main.go

```diff
@@ -1,590 +1,7 @@
-}
-
-func (t *{{.ResourceName}}) SetStatus(_ model.ResourceStatus) error {
-	return errors.New("status not supported")
-}
-
-func (t *{{.ResourceName}}) Descriptor() model.ResourceTypeDescriptor {
-	return {{.ResourceName}}TypeDescriptor 
-}
-{{- if and (hasSuffix .ResourceType "Overview") (ne $baseType "Service") }}
-
-func (t *{{.ResourceName}}) SetOverviewSpec(resource model.Resource, insight model.Resource) error {
-	t.SetMeta(resource.GetMeta())
-	overview := &{{$pkg}}.{{.ProtoType}}{
-		{{$baseType}}: resource.GetSpec().(*{{$pkg}}.{{$baseType}}),
-	}
-	if insight != nil {
-		ins, ok := insight.GetSpec().(*{{$pkg}}.{{$baseType}}Insight)
-		if !ok {
-			return errors.New("failed to convert to insight type '{{$baseType}}Insight'")
-		}
-		overview.{{$baseType}}Insight = ins
-	}
-	return t.SetSpec(overview)
-}
-{{- end }}
-
-var _ model.ResourceList = &{{.ResourceName}}List{}
-
-type {{.ResourceName}}List struct {
-	Items      []*{{.ResourceName}}
-	Pagination model.Pagination
-}
-
-func (l *{{.ResourceName}}List) GetItems() []model.Resource {
-	res := make([]model.Resource, len(l.Items))
-	for i, elem := range l.Items {
-		res[i] = elem
-	}
-	return res
-}
-
-func (l *{{.ResourceName}}List) GetItemType() model.ResourceType {
-	return {{.ResourceType}}Type
-}
-
-func (l *{{.ResourceName}}List) NewItem() model.Resource {
-	return New{{.ResourceName}}()
-}
-
-func (l *{{.ResourceName}}List) AddItem(r model.Resource) error {
-	if trr, ok := r.(*{{.ResourceName}}); ok {
-		l.Items = append(l.Items, trr)
-		return nil
-	} else {
-		return model.ErrorInvalidItemType((*{{.ResourceName}})(nil), r)
-	}
-}
-
-func (l *{{.ResourceName}}List) GetPagination() *model.Pagination {
-	return &l.Pagination
-}
-
-func (l *{{.ResourceName}}List) SetPagination(p model.Pagination) {
-	l.Pagination = p
-}
-
-var {{.ResourceName}}TypeDescriptor = model.ResourceTypeDescriptor{
-		Name: {{.ResourceType}}Type,
-		Resource: New{{.ResourceName}}(),
-		ResourceList: &{{.ResourceName}}List{},
-		ReadOnly: {{.WsReadOnly}},
-		AdminOnly: {{.WsAdminOnly}},
-		Scope: {{if .Global}}model.ScopeGlobal{{else}}model.ScopeMesh{{end}},
-		{{- if ne .KdsDirection ""}}
-		KDSFlags: {{.KdsDirection}},
-		{{- end}}
-		WsPath: "{{.WsPath}}",
-		KumactlArg: "{{.KumactlSingular}}",
-		KumactlListArg: "{{.KumactlPlural}}",
-		AllowToInspect: {{.AllowToInspect}},
-		IsPolicy: {{.IsPolicy}},
-		SingularDisplayName: "{{.SingularDisplayName}}",
-		PluralDisplayName: "{{.PluralDisplayName}}",
-		{{- if ne .ShortName "" }}
-		ShortName: "{{.ShortName}}",{{- end}}
-		IsExperimental: {{.IsExperimental}},
-        IsProxy: {{.IsProxy}},
-{{- if .HasInsights}}
-		Insight: New{{.ResourceType}}InsightResource(),
-		Overview: New{{.ResourceType}}OverviewResource(),
-{{- end}}
-	}
-
-{{- if not .SkipRegistration}}
-func init() {
-	registry.RegisterType({{.ResourceName}}TypeDescriptor)
-}
-{{- end}}
-{{end}}
-`))
-
-var rootDir = "."
+import "github.com/kumahq/kuma/tools/resource-gen/pkg/generator"
```

Comment:
this got moved but GH does not like me:

https://github.com/kumahq/kuma/pull/12680/commits/7356ba3b9edc9d5097dbc33ecfe99ba945292622

Reply:
rendered diff:

https://gistpreview.github.io/?6c8ae4e6a699c83e721dcd8457e26cb2